opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -338,12 +365,10 @@ namespace Opm {
             return static_cast<const Implementation&>(*this);
         }
 
-        // return true if wells are available in the reservoir
-        bool wellsActive() const { return wells_active_; }
```

Comment:
For the `wells()`, we return `*wells_`. I think it is good to keep the assertion there. 

And later, for the `wellsActive`, we probably need to check two categories of wells later (Not sure yet.). I think it might be good to keep this one in the `StandardWells` for now.  

Sorry if I misunderstood you.